src/main/java/uk/gov/companieshouse/api/strikeoffobjections/processor/ObjectionProcessor.java

```diff
@@ -0,0 +1,90 @@
+package uk.gov.companieshouse.api.strikeoffobjections.processor;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.strikeoffobjections.common.ApiLogger;
+import uk.gov.companieshouse.api.strikeoffobjections.common.LogConstants;
+import uk.gov.companieshouse.api.strikeoffobjections.exception.InvalidObjectionStatusException;
+import uk.gov.companieshouse.api.strikeoffobjections.exception.ObjectionNotFoundException;
+import uk.gov.companieshouse.api.strikeoffobjections.model.entity.Objection;
+import uk.gov.companieshouse.api.strikeoffobjections.model.entity.ObjectionStatus;
+import uk.gov.companieshouse.api.strikeoffobjections.service.IObjectionService;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Processes an Objection
+ *
+ * Will only process Objection if status = SUBMITTED
+ * Calls Chips to place stop against company
+ * Sends email
+ */
+@Component
+public class ObjectionProcessor {
+
+    private static final String INVALID_START_STATUS_MSG =
+            "Objection %s has status %s. Cannot process unless status = SUBMITTED";
```

Comment:
This message looks wrong, shouldn't it be 'unless status != SUBMITTED' based on logic check lower down?

Reply:
I'm going to document the status flow that should help clear it up. In the meantime, I think it should be ok as is, the flow should go open->submitted->processing->chips_sent->processed.  We could have email_sent after chips_sent but nothing happens between email_sent and going to processed, so it seems better to go straight to processed which will be the finish state. I'm trying to draw up a finite state machine for it!